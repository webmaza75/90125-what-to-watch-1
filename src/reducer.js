import ActionTypes from './action-types.js';
import filmList from './mocks/films.js';
import {ALL_GENRES} from './consts.js';

const initialState = {
  films: filmList,
  filter: ALL_GENRES,
  filmsGroup: filmList
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.CHANGE_GENRE:
      return {
        ...state,
        filter: action.payload
      };
    case ActionTypes.GET_FILMS_BY_GENRE:
      if (action.payload.filter === ALL_GENRES) {
        return initialState;
      }
      return {
        ...state,
        filmsGroup: action.payload.films.filter(({genre}) => genre.includes(action.payload.filter))
      };
  }
  return state;
};

export default reducer;
