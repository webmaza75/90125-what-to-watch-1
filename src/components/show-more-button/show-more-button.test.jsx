import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';

import ShowMoreButton from './show-more-button.jsx';
import reducer from '../../reducers/reducer.js';

it(`ShowMoreButton correctly renders`, () => {
  const showMoreButton = renderer
    .create(<Provider store={createStore(reducer)}>
      <ShowMoreButton />
    </Provider>)
    .toJSON();

  expect(showMoreButton).toMatchSnapshot();
});
