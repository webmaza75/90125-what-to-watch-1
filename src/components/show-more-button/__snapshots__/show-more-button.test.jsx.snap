// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton correctly renders 1`] = `null`;
